src/Microsoft.AspNetCore.Blazor/Components/InjectAttribute.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Blazor.Components
+{
+    /// <summary>
+    /// Indicates that the associated property should have a value injected from the
+    /// service provider during initialization.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Property, AllowMultiple = false)]
+    public class InjectAttribute : Attribute
```

Comment:
Will you want the `@inject` directive also?

I'm doing some of the directive stuff now so it will be easy to add.

Reply:
Ah, I just did implement that before I saw your comment: https://github.com/aspnet/Blazor/pull/151/commits/f57975118d0766e751ae70a4bb830948ee4f41b4#diff-c3f8f3bf88777effa8c797c34024165d

Does that look OK to you? I know it doesn't support multiple CodeTargets, but I'm not sure whether we need that (and it's not hard to add if so).

Reply:
I also commented elsewhere. Looks good to me.